marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -44,88 +44,172 @@ describe('RTL Source-to-entity map tests', () => {
         expect(getByText("concat(name,'-NEW')")).toBeInTheDocument();
     });
 
-    test('Filtering Name column in Source and Entity tables',() => {
+    test('Filtering Name column in Source (JSON Source Data) and Entity tables', () => {
 
-      const { getByText,getByTestId,queryByText } = render(<SourceToEntityMap {...data.mapProps}
-        mappingVisible={true}
+        const { getByText, getByTestId, queryByText } = render(<SourceToEntityMap {...data.mapProps}
+            mappingVisible={true}
+            sourceData={data.jsonSourceDataMultipleSiblings}
+            entityTypeProperties={data.entityTypePropertiesMultipleSiblings}
         />);
 
-      //Expanding all the nested levels first to apply filter later
-      fireEvent.click(getByTestId('expandCollapseBtn-source'));
-      fireEvent.click(getByTestId('expandCollapseBtn-entity'));
+        /* Test filter for JSON Source data in Source table  */
+        let filterIcon = getByTestId('filterIcon-key');
+        expect(filterIcon).toBeInTheDocument();
+        fireEvent.click(filterIcon);
+        let inputSearch = getByTestId('searchInput-key');
```

Comment:
lets initialize resetSearch here. all the elements here, to go with arrange/act/assert way of writing tests.

Reply:
Done.